### DISCUSSION TITLE
Rationale behind running lighthouse for localhost environment

### DISCUSSION THREAD

**Discussion by himanshuara** [GoogleChrome/lighthouse-ci#923] [General]
I am trying to understand how running lighthouse as CI helps me. What I understood is on every PR lighthouse will run on my application that runs on a localhost. But my end users are using the prod environment which is drastically different from localhost and hence the performance scores will be different too. Also i wont be monitoring the exact experience what the end users will be having. An angular app runs on JIT when running on localhost (dev) and turns on the AOT only in prod env. How are we expecting lighthouse CI to give us exact performance scores. Seeking some counter suggestions.

**Comment by adamraine** [comment]
Worth pointing out that Lighthouse uses [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md) to simulate a network connection that is slower than localhost.

> An angular app runs on JIT when running on localhost (dev) and turns on the AOT only in prod env. How are we expecting lighthouse CI to give us exact performance scores. Seeking some counter suggestions.

Using JIT instead of AOT will probably change your performance scores. The best solution is to run Lighthouse on the same experience that your users will see. In this case that would mean using Lighthouse on the page compiled with AOT.

Please see https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/troubleshooting.md#the-results-im-seeing-dont-match-the-results-i-see-in-another-environment for more information.
